ios/RNZendeskChat.m

```diff
@@ -69,14 +60,46 @@ @implementation RNZendeskChat
     [self showTicketsFunction];
   });
 }
-
 RCT_EXPORT_METHOD(showHelpCenter:(NSDictionary *)options) {
   [self setVisitorInfo:options];
   dispatch_sync(dispatch_get_main_queue(), ^{
     [self showHelpCenterFunction:options];
   });
 }
-
+NSMutableString* mutableLog;
+NSNumber* logId;
+NSMutableDictionary* customFields;
+#ifndef MAX_LOG_LENGTH
+#define MAX_LOG_LENGTH 60000
+#endif
+- (void) initGlobals
+{
+    if(mutableLog == nil){
+        mutableLog = [[NSMutableString alloc] init];
+    }
+    if(logId == nil){
+        logId = [NSNumber numberWithLong: 4413278012049];
+    }
+}
+RCT_EXPORT_METHOD(appendLog:(NSString *)log) {
+    [self initGlobals];
+    [mutableLog insertString:log atIndex:0];
+    [mutableLog substringToIndex:fmax(0,fmin(MAX_LOG_LENGTH,mutableLog.length))];
+}
+- (void) addTicketCustomFieldFunction:(NSString *)key withValue:(NSString *)value
+{
+    if(customFields == nil){
+        customFields = [[NSMutableDictionary alloc] init];
+    }
```

Comment:
```suggestion
    if(customFields == nil){
        customFields = [[NSMutableDictionary alloc] init];
    }
```
Maybe could we move this init `initGlobals`?

Reply:
Yes, it's a good idea! Moved in [ab03762](https://github.com/pagopa/io-react-native-zendesk/pull/10/commits/ab03762f7e313302a55bb06cd58402ff1ab1d319)